File: 02. Programming - Java/Test/src/test/Shape.java
package test;

public abstract class Shape {

	protected int a;
	
	public Shape(int a) {
		this.a=a;
	}

	public abstract double calculateArea();
	
	public int getA() {
		return a;
	}

	public void setA(int a) {
		this.a = a;
	}

	
}

File: 03. Programming - C#/TestPreparation5Example/Shape.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TestPreparation5Example
{
    abstract class Shape
    {
        private string type;
        public string Type
        {
            get
            {
                return type;
            }
            set
            {
                type = value;
            }
        }
        private double a;
        public double A
        {
            get
            {
                return a;
            }set => a = value;
        }

        public Shape(string type,double a)
        {
            this.type = type;
            this.a = a;
        }

        public abstract double calculatePerimeter();
        public abstract double calculateArea();
    }
}
